plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php

```diff
@@ -145,29 +138,31 @@ public function watchFolder(KalturaDropFolder $dropFolder)
 				$nextLink = $items['@odata.deltaLink'];
 			}
 
-			KalturaLog::info("Next page for user $userId: $nextLink");
-			$newDriveUrlAssoc[] = array($userId, $nextLink);
-			$line = fgetcsv($drivesFileHandle);
-		}
-
-		$newFileName = $driveLocationsDir . DIRECTORY_SEPARATOR . $driveLocationsFileNamePrefix . $this->dropFolder->id . self::LAST_TIMESTAMP_POSTFIX . time() . '.csv';
-		KalturaLog::info("Creating new file: $newFileName. Inserting " . count($newDriveUrlAssoc) . ' lines.');
-
-		$newFileHandle = fopen($newFileName, 'w+');
-		foreach ($newDriveUrlAssoc as $line)
-		{
-			fputcsv($newFileHandle, $line);
-		}
-		fclose($newFileHandle);
-		fclose($drivesFileHandle);
-
-		if (!file_exists($newFileName))
-		{
-			KalturaLog::info("New filename $newFileName was NOT created!");
-			KalturaLog::info(print_r($newDriveUrlAssoc, true));
+			KalturaLog::info("Next page for user {$user->id}: $nextLink");
+			$args = explode('&', parse_url($nextLink, PHP_URL_QUERY));
+
+			foreach ($args as $arg)
+            {
+                list($key, $value) = explode('=', $arg);
```

Comment:
this will always be exploded correctly?

Reply:
@inbal-ben-david  yes, but I also added a thrown exception in case nextLink could not be found.